pwiz_tools/Shared/PanoramaClient/PanoramaUtil.cs

```diff
@@ -0,0 +1,1096 @@
+                }
+                else throw;
+            }
+        }
+    
+        public FolderOperationStatus DeleteFolder(string folderPath, string username, string password)
+        {
+            var parentFolderStatus = IsValidFolder(folderPath, username, password);
+            switch (parentFolderStatus)
+            {
+                case FolderState.nopermission:
+                    return FolderOperationStatus.nopermission;
+                case FolderState.notfound:
+                    return FolderOperationStatus.notfound;
+                case FolderState.notpanorama:
+                    return FolderOperationStatus.notpanorama;
+            }
+    
+            try
+            {
+                using (var webClient = new WebClientWithCredentials(ServerUri, username, password))
+                {
+                    Uri requestUri = PanoramaUtil.CallNewInterface(ServerUri, @"core", folderPath, @"deleteContainer", "", true);
+                    JObject result = webClient.Post(requestUri, "");
+                    return FolderOperationStatus.OK;
+                }
+            }
+            catch (WebException ex)
+            {
+                var response = ex.Response as HttpWebResponse;
+                if (response != null && response.StatusCode != HttpStatusCode.OK)
+                {
+                    return FolderOperationStatus.error;
+                }
+                else throw;
+            }
+        }
+    
+        public PanoramaServer EnsureLogin(PanoramaServer server)
+        {
+            var refServerUri = server.URI;
+            UserState userState = PanoramaUtil.ValidateServerAndUser(ref refServerUri, server.Username, server.Password);
+            if (userState.IsValid())
+            {
+                return server.ChangeUri(refServerUri);
+            }
+            else
+            {
+                throw new PanoramaServerException(userState.GetErrorMessage(refServerUri));
+            }
+        }
+    
+        public JToken GetInfoForFolders(PanoramaServer server, string folder)
+        {
+            if (server.HasUserCredentials())
+            { 
+                server = EnsureLogin(server);
+            }
+           
+    
+            // Retrieve folders from server.
+            Uri uri = PanoramaUtil.GetContainersUri(server.URI, folder, true);
+    
+            using (var webClient = new WebClientWithCredentials(server.URI, server.Username, server.Password))
+            {
+                return webClient.Get(uri);
+            }
+        }
+
+        /// <summary>
+        /// Given the name of a file, allows a user to select
+        /// which folder on their machine to download the file to
+        /// and returns the path of the selected folder
+        /// </summary>
+        /// <param name="fileName"></param>
+        /// <param name="lastPath"></param>
+        /// <returns></returns>
+        public string SaveFile(string fileName, string lastPath)

```

Comment:
Since this is not actually saving the file can it be renamed so something like SelectDownloadPath? Have you tried using SaveFileDialog instead of FolderBrowserDialog?